pom.xml

```diff
@@ -2726,6 +2726,26 @@
     <build>
         <pluginManagement>
             <plugins>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-compiler-plugin</artifactId>
+                    <version>3.8.0</version>
+                    <configuration>
+                        <source>8</source>
+                        <target>8</target>
+                        <compilerArgs>
+                            <arg>-XDcompilePolicy=simple</arg>
+                            <arg>-Xplugin:ErrorProne</arg>
```

Comment:
what is the impact on the compile time?

Reply:
```
Without:
mvn package -T0.5C -DskipTests  2116.88s user 120.28s system 681% cpu 5:28.31 total
With:
mvn package -T0.5C -DskipTests  3050.80s user 131.11s system 685% cpu 7:44.38 total
```

Too much, I'll drop it

Reply:
could be out of `compile` stage, we could group that stage with checkstyle

Reply:
I have tried something, I am not sure it is what you want, pairing on it would be great

Reply:
Looking at https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html#lifecycle-reference there's not phase that would fit between compile and verify.

Either we enable it at test stage or at compile stage or we wait the switch to gradle (no ETA at all on that)

Reply:
let's drop it then